src/qibo/ui/mpldrawer.py

```diff
@@ -0,0 +1,537 @@
+                cluster_gates.append(temp_mgates)
+                temp_mgates = []
+
+            if "MEASURE" not in item[0]:
+                cluster_gates.append([item])
+        i = i + 1
+
+    if len(temp_gates) > 0:
+        cluster_gates.append(temp_gates)
+
+    if len(temp_mgates) > 0:
+        cluster_gates.append(temp_mgates)
+
+    temp_gates = []
+    temp_mgates = []
+
+    return cluster_gates
+
+
+def _set_style(style):
+
+    if style is None:
+        default_values = dict(
+            scale=1.0,
+            fontsize=14.0,
+            linewidth=1.0,
+            control_radius=0.05,
+            not_radius=0.15,
+            swap_delta=0.08,
+            label_buffer=0.0,
+            facecolor="w",
+            edgecolor="#000000",
+            fillcolor="#000000",
+            linecolor="k",
+            textcolor="k",
+            gatecolor="w",
+            controlcolor="#000000",
+        )
+
+        plot_params.update(default_values)
+    else:
+        if type(style) is str:
+            plot_params.update(_get_style(style))
+        elif type(style) is dict:
+            plot_params.update(style)
+
+
+def _plot_qibo_circuit(circuit, scale, cluster_gates, style):
+
+    _set_style(style)
+
+    inits = list(range(circuit.nqubits))
+
+    labels = []
+    for i in range(circuit.nqubits):
+        labels.append("q_" + str(i))
+
+    if len(circuit.queue) > 0:
+        gates_plot = []
+
+        for gate in circuit.queue:
+            init_label = gate.name.upper()
+            if "CCX" in init_label:
+                init_label = "TOFFOLI"
+
+            if "CX" in init_label:
+                init_label = "CNOT"
+
+            if (
+                len(gate._control_qubits) > 0
+                and "C" in init_label[0]
+                and "CNOT" not in init_label
+            ):
+                init_label = init_label[1:]
+
+            item = ()
+            item += (init_label,)
+
+            for qbit in gate._target_qubits:
+                if qbit is tuple:
+                    item += ("q_" + str(qbit[0]),)
+                else:
+                    item += ("q_" + str(qbit),)
+
+            for qbit in gate._control_qubits:
+                if qbit is tuple:
+                    item += ("q_" + str(qbit[0]),)
+                else:
+                    item += ("q_" + str(qbit),)
+
+            gates_plot.append(item)
+
+        if cluster_gates:
+            gates_cluster = _make_cluster_gates(gates_plot)
+            return _plot_quantum_schedule(gates_cluster, inits, labels, scale=scale)
+
+        return _plot_quantum_circuit(gates_plot, inits, labels, scale=scale)
+    else:
+        return _plot_lines_circuit(inits, labels, scale=scale)
+
+
+def plot(circuit, scale=0.6, cluster_gates=True, style=None, save_file=None):
+
+    ax = _plot_qibo_circuit(circuit, scale, cluster_gates, style)
+
+    if save_file:
+        ax.figure.savefig(save_file, bbox_inches="tight")
```

Comment:
As in the previous comment: this part could be completely omitted, and left up to the user (you're trading one function call for one parameter).

And without this, the `plot` function would just be the function `_plot_qibo_circuit`. So just rename that function :)

Reply:
thanks! I removed the function and leave the option to the user